drivers/sensor/icp10125/icp10125.c

```diff
@@ -194,29 +190,26 @@ static int icp10125_measure(const struct i2c_dt_spec *i2c, const struct icp10125
 
 	rc = i2c_write_dt(i2c, (uint8_t *)&cmds[mode], sizeof(cmds[mode]));
 	if (rc < 0) {
-		LOG_ERR("Failed to start measurement.\n");
+		LOG_ERR("Failed to start measurement.");
 		return rc;
 	}
 
 	/* Wait for the sensor to become readable.
 	 * First wait for the typical time and then read.
 	 * If that fails, wait until the time to surely became readable.
 	 */
-	k_sleep(K_USEC(conv_time_typ[mode]));
-	if (i2c_read_dt(i2c, (uint8_t *)sensor_data, sizeof(sensor_data[0]) * data_num) < 0) {
-		k_sleep(K_USEC(conv_time_max[mode] - conv_time_typ[mode]));
-		rc = i2c_read_dt(i2c, (uint8_t *)sensor_data, sizeof(sensor_data[0]) * data_num);
-		if (rc < 0) {
-			LOG_ERR("Failed to read measurement.\n");
-			return rc;
-		}
+	k_sleep(K_USEC(conv_time_max[mode]));
+	rc = i2c_read_dt(i2c, (uint8_t *)sensor_data, sizeof(sensor_data[0]) * data_num);
+	if (rc < 0) {
+		LOG_ERR("Failed to read measurement.");
+		return rc;
 	}
 
 #ifdef CONFIG_ICP10125_CHECK_CRC
 	/* Calculate CRC from Chapter 5 Section 8 of ICP10125 Product manuals. */
 	for (size_t i = 0; i < data_num; i++) {
 		if (!icp10125_check_crc(sensor_data[i].data, SENSOR_DATA_SIZE)) {
```

Comment:
As discussed on Teams, there's a bug here, predating our changes. I believe that the fix is to change `!icp10125_check_crc(sensor_data[i].data, SENSOR_DATA_SIZE)`  to `icp10125_check_crc(sensor_data[i].data, SENSOR_DATA_SIZE) != sensor_data[i].crc`.

Reply:
changed and tested